src/toil/provisioners/gceProvisioner.py

```diff
@@ -0,0 +1,915 @@
+        commandTokens.append('%s@%s' % (user,str(nodeIP)))
+        appliance = kwargs.pop('appliance', None)
+        if appliance:
+            # run the args in the appliance
+            tty = kwargs.pop('tty', None)
+            ttyFlag = '-t' if tty else ''
+            commandTokens += ['docker', 'exec', '-i', ttyFlag, 'toil_leader']
+        inputString = kwargs.pop('input', None)
+        if inputString is not None:
+            kwargs['stdin'] = subprocess.PIPE
+        collectStdout = kwargs.pop('collectStdout', None)
+        if collectStdout:
+            kwargs['stdout'] = subprocess.PIPE
+        kwargs['stderr'] = subprocess.PIPE
+        logger.debug('Node %s: %s', nodeIP, ' '.join(args))
+        args = list(map(pipes.quote, args))
+        commandTokens += args
+        logger.debug('Full command %s', ' '.join(commandTokens))
+        popen = subprocess.Popen(commandTokens, **kwargs)
+        stdout, stderr = popen.communicate(input=inputString)
+        # at this point the process has already exited, no need for a timeout
+        resultValue = popen.wait()
+        # ssh has been throwing random 255 errors - why?
+        if resultValue != 0:
+            logger.info('SSH Error (%s) %s' % (resultValue, stderr))
+            raise RuntimeError('Executing the command "%s" on the appliance returned a non-zero '
+                               'exit code %s with stdout %s and stderr %s'
+                               % (' '.join(args), resultValue, stdout, stderr))
+        return stdout
+
+
+    def _terminateInstances(self, instances):
+        def worker(driver, instance):
+            logger.info('Terminating instance: %s', instance.name)
+            driver.destroy_node(instance)
+
+        driver = self._getDriver()
+        threads = []
+        for instance in instances:
+            t = threading.Thread(target=worker, args=(driver,instance))
+            threads.append(t)
+            t.start()
+
+        logger.info('... Waiting for instance(s) to shut down...')
+        for t in threads:
+            t.join()
+
+    DEFAULT_TASK_COMPLETION_TIMEOUT = 180
+    def ex_create_multiple_nodes(
+            self, base_name, size, image, number, location=None,
+            ex_network='default', ex_subnetwork=None, ex_tags=None,
+            ex_metadata=None, ignore_errors=True, use_existing_disk=True,
+            poll_interval=2, external_ip='ephemeral',
+            ex_disk_type='pd-standard', ex_disk_auto_delete=True,
+            ex_service_accounts=None, timeout=DEFAULT_TASK_COMPLETION_TIMEOUT,
+            description=None, ex_can_ip_forward=None, ex_disks_gce_struct=None,
+            ex_nic_gce_struct=None, ex_on_host_maintenance=None,
+            ex_automatic_restart=None, ex_image_family=None,
+            ex_preemptible=None):
+        """
+         Monkey patch to gce.py in libcloud to allow disk and images to be specified.
+         Also changed name to a uuid below.
+        """
```

Comment:
Could this docstring include comments about what all of these variables are?

Reply:
No. This is a monkey patch.